include/solvers/mpm_base.tcc

```diff
@@ -92,6 +92,39 @@ mpm::MPMBase<Tdim>::MPMBase(const std::shared_ptr<IO>& io) : mpm::MPM(io) {
     abort();
   }
 
+  // VTK particle variables
+  try {
+    // Initialise container with empty vector
+    vtk_vars_.insert(
+        std::make_pair(mpm::VariableType::Scalar, std::vector<std::string>()));
+    vtk_vars_.insert(
+        std::make_pair(mpm::VariableType::Vector, std::vector<std::string>()));
+    vtk_vars_.insert(
+        std::make_pair(mpm::VariableType::Tensor, std::vector<std::string>()));
+
+    if (post_process_.at("vtk").is_array() &&
+        post_process_.at("vtk").size() > 0) {
+      // Iterate over vtk
+      for (unsigned i = 0; i < post_process_.at("vtk").size(); ++i) {
+        std::string attribute =
+            post_process_["vtk"][i].template get<std::string>();
+        if (mpm::variables.find(attribute) != mpm::variables.end())
+          vtk_vars_[mpm::variables.at(attribute)].emplace_back(attribute);
+        else {
+          console_->warn(
+              "{} #{}: VTK variable \"{}\" was specified, but is not available "
+              "in variable list",
+              __FILE__, __LINE__, attribute);
+        }
+      }
+    } else {
```

Comment:
There is no reason this should be a runtime error. Use console warn instead.

Reply:
I just followed the one in `state_var` and other input functions with `try` and `catch` just to make it the same.

Reply:
Could you change it to console->warn, it shound't throw.

Reply:
Should I change the other part too in this PR or the next one?

Reply:
Both please

Reply:
@kks32 Just to confirm, you want me to change all the catch and throw here too?
https://github.com/cb-geo/mpm/blob/2481028eb09f0cfd7da91bd27dfb860dc5dff272/include/solvers/mpm_base.tcc#L24-L93

Reply:
No need to change those. Just change vtk_statevariable

Reply:
Okay, I will make the change for the vtk and the vtk_statevars in PR #683